src/mlpack/core/data/image_info.hpp

```diff
@@ -28,12 +27,10 @@
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #include <stb_image_write.h>
 
-#endif

```

Comment:
Any STB-related code---including the `#include`s---should be in the .cpp file you made and removed in all other files.  So these includes between the `#ifdef` that you removed should also be removed.